.github/workflows/deployment.yml

```diff
@@ -108,6 +108,19 @@ jobs:
         run: |
           shopt -s failglob
           script/sign dist/gh_*_macOS_*.zip
+      - name: Build universal macOS pkg installer
+        if: inputs.environment != 'production'
+        env:
+          TAG_NAME: ${{ inputs.tag_name }}
+        run: script/pkgmacos "$TAG_NAME"
```

Comment:
**How have you been testing `script/pkgmacos` locally?**

I was wondering how large the new `.pkg` files were, so tried building it locally.  I thought about trying to build this locally only to realize we probably want to add a new target to `Makefile` as this involves multiple steps.

Here's how far I've got so far before realizing I should simply ask 😅 

```shell
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ ./script/pkgmacos 
To build a universal pkg for macOS:
  script/pkgmacos <tag-name>

To build and sign set APPLE_DEVELOPER_INSTALLER_ID environment variable before.
For example, if you have a signing identity with the identifier 
"Developer ID Installer: Your Name (ABC123DEF)" set it in the variable.
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ ./script/pkgmacos v0.0.0
build_pkg:7: no matches found: ./share/man/man1/gh*.1
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ vim Makefile 
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ make
go build -trimpath -ldflags "-X github.com/cli/cli/v2/internal/build.Date=2024-05-23 -X github.com/cli/cli/v2/internal/build.Version=v2.49.2-50-g9454d5e7 " -o bin/gh ./cmd/gh
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ ./script/pkgmacos v0.0.0
build_pkg:7: no matches found: ./share/man/man1/gh*.1
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ vim Makefile 
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ make manpages
go run ./cmd/gen-docs --man-page --doc-path ./share/man/man1/
andyfeller@Andys-MBP:cli/cli ‹feature-macos-pkg-installer›$ ./script/pkgmacos v0.0.0
```

Reply:
Concerning the pkg size, its about 24MB:
`-rw-r--r--@ 1 paulober  staff    24M May 24 14:10 gh_v2.99.8_macOS_universal.pkg`

To test it locally:
```zsh
make manpages
make completions
./script/release --local "v2.99.8" --platform macos
./script/pkgmacos v2.99.8
```

Reply:
I take that back, it wasn't that difficult. You can now build a pkg using `make macospkg VERSION=v2.99.8`. (goreleaser required)

Reply:
this is really going to help the team, thank you for saving us future time 🫶